modules/@apostrophecms/i18n/index.js

```diff
@@ -818,6 +818,393 @@ module.exports = {
+          relatedOnly,
+          log = [],
+          seen = new Set()
+        }) {
+          const docs = [];
+          if (seen.has(doc.aposDocId)) {
+            return log;
+          }
+          seen.add(doc.aposDocId);
+          if (!canLocalize(req, doc, false)) {
+            log.push({
+              _id: doc._id,
+              aposDocId: doc.aposDocId,
+              type: doc.type,
+              title: doc.title,
+              relationship: false,
+              error: true
+            });
+            self.logError(
+              req,
+              'localize-batch-can-localize-error',
+              'The document type can\'t be localized or insufficient permissions',
+              {
+                id: doc._id,
+                type: doc.type,
+                title: doc.title,
+                relationship: false
+              }
+            );
+            if (reporting) {
+              reporting.failure();
+            }
+            return log;
+          }
+          if (!relatedOnly) {
+            docs.push(doc);
+          }
+
+          if (relatedTypes.size > 0) {
+            try {
+              await findRelatedDocs(req.clone({ mode: 'draft' }), {
+                doc,
+                schema: self.apos.modules[doc.type].schema,
+                relatedTypes,
+                memo: docs,
+                seen
+              });
+            } catch (e) {
+              self.logError(
+                req,
+                'localize-batch-related-error',
+                'Error finding related documents',
+                {
+                  id: doc._id,
+                  type: doc.type,
+                  title: doc.title,
+                  error: e.message,
+                  stack: e.stack.split('\n').slice(1).map(line => line.trim())
+                }
+              );
+              if (reporting) {
+                reporting.failure();
+              }
+              return log;
+            }
+          }
+
+          let hasError = false;
+          for (const item of docs) {
+            const manager = self.apos.doc.getManager(item.type);
+            // Not using Promise.allSettled because of potential
+            // rate limit issues with external services when i.e.
+            // automatically translating content.
+            // Related info:
+            // https://cookbook.openai.com/examples/how_to_handle_rate_limits
+            for (const locale of toLocales) {
+              const payload = {
+                _id: item._id,
+                aposDocId: item.aposDocId,
+                locale,
+                type: item.type,
+                title: item.title,
+                relationship: item.aposDocId !== doc.aposDocId ? doc.aposDocId : false,
+                error: false
+              };
+              try {
+                await manager.localize(req, item, locale, {
+                  update: !payload.relationship ? true : update,
+                  batch: true
```

Comment:
This flag is still not accounted anywhere. The idea is to be passed down to the localization event handlers.